### DISCUSSION TITLE
DirectQuery - How to select subset of columns from table?

### DISCUSSION THREAD

**Discussion by HuifangYeo** [atoti/atoti#799] [Q&A] [answered]
Hi,

In the Java version, we are able to [select a subset of columns from the table](https://docs.activeviam.com/products/atoti/server/latest/docs/directquery/how_to/customize_table/#select-a-subset-of-fields).
Is this possible with the Python version?

**Comment by OPeyrusse** [comment, marked as answer]
If I am not mistaken, if you pass a mapping of columns to `#add_external_table`, it will only retain the columns that are in the mapping
https://docs.atoti.io/latest/lib/atoti/atoti.session.html#atoti.Session.add_external_table

**Reply by OPeyrusse** [reply]
I agree that it is a bit cumbersome if you don't want to rename the columns, though it should be relatively easy to generate
